wagtail_streamfield_migration_toolkit/autodetect/comparers.py

```diff
@@ -138,146 +132,179 @@ def hashable_deep_deconstruct(cls, obj):
 
+    child_option_weight = None
+    base_option_weight = None
+
     @classmethod
-    def compare_args(cls, old_args, new_args):
-        old_children, new_children = cls.get_children(old_args, new_args)
+    def compare_options(cls, old_options, new_options):
+        old_children, new_children, old_options, new_options = cls.get_children(
+            old_options, new_options
+        )
 
-        child_score_sum = 0
-        new_children_by_name = {
-            new_child_name: new_child_tuple
-            for new_child_name, new_child_tuple in new_children
-        }
-        for old_child_name, (
-            old_child_path,
-            old_child_args,
-            old_child_kwargs,
-        ) in old_children:
-            old_child_def = import_class(old_child_path)
-            comparer: BaseBlockDefComparer = (
-                block_def_comparer_registry.get_block_def_comparer_for_class(
-                    old_child_def
-                )
-            )
-            # TODO do a process similar to what we do outside, where we rank blocks and pick one
-            # to map to.
-            # TODO problem: what if we map one block, but then there would be another block which
-            # we come across afterwards that would have better mapped to it? Also if we're unsure,
-            # do we simply leave blocks unmapped altogether?
-            if old_child_name in new_children_by_name:
-                new_child_path, new_child_args, new_child_kwargs = new_children_by_name[
-                    old_child_name
-                ]
-                child_score_sum += comparer._compare(
-                    old_name=old_child_name,
-                    old_path=old_child_path,
-                    old_args=old_child_args,
-                    old_kwargs=old_child_kwargs,
-                    new_name=old_child_name,
-                    new_path=new_child_path,
-                    new_args=new_child_args,
-                    new_kwargs=new_child_kwargs,
-                )
-
-        return child_score_sum / len(old_children)
+        old_child_dict = dict(old_children)
+        new_child_dict = dict(new_children)
 
-    @staticmethod
-    def get_children(old_args, new_args):
-        """Returns two lists, children of the old block and children of the new block.
+        # TODO do proper mapping of children, not just by name. Currently we have a very simplistic
+        # mapping by name.
 
-        `old_children, new_children = cls.get_children(old_args, new_args)`
-        """
-        raise NotImplementedError
+        mapped_child_names = list(
+            set(old_child_dict.keys()).intersection(set(new_child_dict.keys()))
+        )
+        all_child_names = list(
+            set(old_child_dict.keys()).union(set(new_child_dict.keys()))
+        )
+
+        child_score = len(mapped_child_names) / len(all_child_names)
+        option_score = super().compare_options(old_options, new_options)
```

Comment:
It looks like children aren't removed from `old_options` and `new_options` before passing them here, so we've got an extra child comparison in the `compare_options` method. Is that intended?